Objects/object.c

```diff
@@ -2150,6 +2155,9 @@ _PyTrash_thread_destroy_chain(void)
 int
 _PyTrash_begin(PyThreadState *tstate, PyObject *op)
 {
```

Comment:
Can you please insert an assertion to ensure that the object implement the GC protocol?